components/push/src/communications.rs

```diff
@@ -259,36 +261,58 @@ impl Connection for ConnectHttp {
     ///
     /// A `None` channel_id indicates that we are unsubscribing
     /// from all channels
-    fn unsubscribe(&mut self, channel_id: Option<&str>) -> error::Result<bool> {
+    fn unsubscribe(&mut self, channel_id: &str) -> error::Result<bool> {
         if self.auth.is_none() {
             return Err(CommunicationError("Connection is unauthorized".into()).into());
         }
         if self.uaid.is_none() {
             return Err(CommunicationError("No UAID set".into()).into());
         }
         let options = self.options.clone();
-        let mut url = format!(
-            "{}://{}/v1/{}/{}/registration/{}",
+        let url = format!(
+            "{}://{}/v1/{}/{}/registration/{}/subscription/{}",
             &options.http_protocol.unwrap(),
             &options.server_host,
             &options.bridge_type.unwrap(),
             &options.sender_id,
             &self.uaid.clone().unwrap(),
+            channel_id,
         );
-        if let Some(channel_id) = channel_id {
-            url = format!("{}/subscription/{}", url, channel_id)
-        }
         if &self.options.sender_id == "test" {
             return Ok(true);
         }
         let response = Request::delete(Url::parse(&url)?)
             .headers(self.headers()?)
             .send()?;
         self.check_response_error(&response)?;
-        if channel_id.is_none() {
-            self.uaid = None;
-            self.auth = None;
+        Ok(true)
+    }
+
+    fn unsubscribe_all(&mut self) -> error::Result<bool> {
+        if self.auth.is_none() {
+            return Err(CommunicationError("Connection is unauthorized".into()).into());
+        }
+        if self.uaid.is_none() {
+            return Err(CommunicationError("No UAID set".into()).into());
+        }
+        let options = self.options.clone();
+        let url = format!(
+            "{}://{}/v1/{}/{}/registration/{}",
+            &options.http_protocol.unwrap(),
+            &options.server_host,
+            &options.bridge_type.unwrap(),
+            &options.sender_id,
+            &self.uaid.clone().unwrap(),
+        );
+        if &self.options.sender_id == "test" {
```

Comment:
Is this only used by rust tests? If so, maybe wrapping it in a `cfg` check would be good? (Ditto for the above which existed before this patch)

Reply:
Looks like it's used by the Kotlin tests 😞 This could be a thing to keep a close eye on when `uniffing` the component though!